### PR TITLE
Migrate APIM to `sku` block, and add zones and public IP support

### LINKED ISSUE #12021: Support for specifying public IP for Azure API Management
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I tried to edit the public IP for an Azure API Management Instance through Terraform, and it looks like the public IP is not an editable field currently. I see it’s available as an output variable, but not an input variable. 
I tried a few times to set it as shown below. 

This was an Azure APIM instance that I had previously created so was trying to edit the public IP setting for it; currently, you can edit this in the Azure portal, PowerShell, CLI, etc. 

The error I received was: "This field cannot be set"

### New or Affected Resource(s)

* azurerm_api_management

### Potential Terraform Configuration

```hcl
virtual_network_type = "External"

virtual_network_configuration {
subnet_id = data.azurerm_subnet.example.id
}
additional_location {
location = "East US"
public_ip_addresses = var.public_ips
}
```

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/api_management

* #0000

### LINKED ISSUE #13445: r/api_management - support for publicIpAddressId for multi-zone deployments
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.13.4

Your version of Terraform is out of date! The latest version
is 1.0.7. You can update by downloading from https://www.terraform.io/downloads.html
```

```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=2.76.0"
    }
  }
}
```
### Affected Resource(s)

* `azurerm_api_management `

### Terraform Configuration Files

```hcl
resource "azurerm_api_management" "main" {
  name                = var.api_management_name
  resource_group_name = var.resource_group_name
  location            = var.api_management_location

  sku_name        = "Premium_3"
  publisher_name  = var.api_management_publisher_name
  publisher_email = var.api_management_publisher_email

  zones = [1,2,3]
  virtual_network_type = var.virtual_network_type

  dynamic "virtual_network_configuration" {
    for_each = var.apim_locations != null ? slice(var.apim_locations, 0, 1) : []
    content {
      subnet_id = data.azurerm_subnet.main[0].id

    }
  }
  dynamic "additional_location" {
    for_each = var.apim_locations != null && var.api_management_sku_name == "Premium_1" ? slice(var.apim_locations, 1, length(var.apim_locations)) : []
    content {
      location = lookup(additional_location.value, "location", null)
      dynamic "virtual_network_configuration" {
        for_each = lookup(additional_location.value, "subnet_name", null) == null ? [] : [1]
        content {
          subnet_id = data.azurerm_subnet.main[index(var.apim_locations, additional_location.value)].id
        }
      }
    }
  }
  tags = var.api_management_tags
}
```

### Actual Behaviour

From the terraform documentation, I was specifying the availiablity zone 1,2,3 and terraform through me the error,
```
{\"error\":{\"code\":\"NotSupported\",\"message\":\"Deploying Microsoft.ApiManagement/service into      
Virtual Network in location `Australia East` and configuring Availability Zones (1,2,3) is only supported  
 in api-version 2021-01-01-preview,2021-04-01-preview,2021-08-01 and setting property   
`publicIpAddressId` is a must. Refer to aka.ms/azurestandardpublicip on how to create a   
Standard SKU Public IP Address resource.\",\"details\":null,\"innerError\":null}}"
```
there is no option to specify the public IP address ID in the terraform resource `azurerm_api_management`

### Steps to Reproduce

1. `terraform apply`

### PR DESCRIPTION
**Pull request by oWretch**
This PR adds the functionality released in the 2021-08-01 API update to the core API Management resources.

Resolves #12021
Resolves #13445

### CODE REVIEW

**Comment by lrsvendsen** [comment]
Any updates on this PR? The added functionality is is really needed.

**Comment by katbyte** [comment]
Thanks for the pr @oWretch - however we try to use `sku_name` instead of a sku block unless absolutely necessary. Could we undo that change? and add capacity as a top level property? thanks!

**Comment by oWretch** [comment]
Thanks @katbyte & @tombuildsstuff. I’m happy revert that change if required. 

My reason for making the change was to keep consistency through the main and additional locations. It is now supported to run a different capacity for the main location and for the individual additional locations. I added the `capacity` property to the `additional_location` block and then moved to the `sku` block so we could have a separate `capacity` property there which then resembled the same property in the additional location configuration.

Let me know which way you prefer me to go.

**Comment by lrsvendsen** [comment]
Sorry for being annoying. Any chance to get it into 2.92?

@oWretch - Thanks for the good work!

**Comment by katbyte** [comment]
@oWretch - i think what we can do is keep the original `sku_name` property and then have a capacity in the additional location block that overrides the default set by the sku name?

**Comment by oWretch** [comment]
> @oWretch - i think what we can do is keep the original `sku_name` property and then have a capacity in the additional location block that overrides the default set by the sku name?

@katbyte will do. Won’t be for a couple of weeks as I have limited connectivity for the moment.

**Comment by Khayoann2** [comment]
Hello,

Any news about this PR ?

Thank you

File: internal/services/apimanagement/api_management_resource.go [hunk 1]
```diff
@@ -544,6 +559,22 @@ func resourceApiManagementService() *pluginsdk.Resource {
 				},
 			},
 
+			"public_ip_address_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"public_network_access": {
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_resource.go:568]: Could we change this to be a boolean and match the rest of the provider?
```suggestion
			"public_network_access_enabled": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @oWretch - overall this looks good aside from one comment about public network access, elsewhere in the provider we change it to a bool and add enabled into the name. could we match that here? 

https://github.com/hashicorp/terraform-provider-azurerm/blob/817593a715edc0f4086008c1bc2f9b7dd5d166e3/internal/services/redis/redis_cache_resource.go#L289

**Comment by katbyte** [comment]
Hope you don't mind @oWretch  but i've pushed that change to get this merged today

**Review by mbfrahry — APPROVED** [review]
LGTM

File: internal/services/apimanagement/api_management_resource.go [hunk 2]
```diff
@@ -17,6 +18,7 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
 	apimValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cognitive/sdk/2021-04-30/cognitiveservicesaccounts"
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_resource.go:21]:
```suggestion
```

**Comment by oWretch** [comment]
> Hope you don't mind @oWretch but i've pushed that change to get this merged today

I don’t mind at all. Thanks for your help to get this committed

**Pull request merged**